src/platform/errors/errorHandler.ts

```diff
@@ -174,6 +180,42 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
             return response === JupyterInterpreterDependencyResponse.ok
                 ? KernelInterpreterDependencyResponse.ok
                 : KernelInterpreterDependencyResponse.cancel;
+        } else if (err instanceof RemoteJupyterServerConnectionError) {
+            const selection = await this.applicationShell.showErrorMessage(
+                DataScience.remoteJupyterConnectionFailedWithServer().format(err.baseUrl),
+                { detail: err.originalError.message || '', modal: true },
+                DataScience.removeRemoteJupyterConnectionButtonText(),
+                DataScience.changeRemoteJupyterConnectionButtonText(),
+                DataScience.selectDifferentKernel()
+            );
+            const serverId = err.serverId;
+            switch (selection) {
```

Comment:
This lets users decide what needs to be done when the remote server is no longer valid.
Fixes #9167